enable-kdump/ubuntu-enable-kdump.yaml

```diff
@@ -0,0 +1,82 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: enable-kdump
+  labels:
+    app: enable-kdump
+spec:
+  selector:
+    matchLabels:
+      name: enable-kdump
+  template:
+    metadata:
+      labels:
+        name: enable-kdump
+    spec:
+      nodeSelector:
+        "cloud.google.com/gke-os-distribution": "ubuntu"
+        "enable-kdump": "true"
+      hostPID: true
+      containers:
+        - name: enable-kdump
+          image: debian
+          imagePullPolicy: Always
+          securityContext:
+            privileged: true
+          command:
+            - /usr/bin/nsenter
+            - -t 1
+            - -m
+            - -u
+            - -i
+            - -n
+            - -p
+            - --
+            - /bin/bash
+            - -c
+            - |
+
+              set -o errexit
+              set -o pipefail
+              set -o nounset
+
+              function check_kdump() {
+                  local kdump_show
+                  kdump_show=$(kdump-config show)
+                  if echo "${kdump_show}" | grep -q "ready to kdump"; then
+                      echo "ready to kdump!"
+                  else
+                      echo "kdump not setup, isn't ready"
+                  fi
+                  echo "${kdump_show}"
+              }
+
+              function install() {
+                  echo "installing kdump"
+                  apt-get update
+                  DEBIAN_FRONTEND=noninteractive apt-get install -y kdump-tools
+                  echo "kdump enabled; rebooting..."
```

Comment:
You also want to enable hung task panic

```
echo 1 > /proc/sys/kernel/hung_task_panic"
```

Reply:
You need to change /etc/sysctl files if you want hung triggered kdump automatically.

Reply:
Oh yeah:

```
echo 1 > /proc/sys/kernel/sysrq
```

Reply:
Thanks good point. Setup the sysctls for `hung_task_panic` and `hung_task_timeout_secs`, and made it so they are applied on next reboot after kdump is installed.

@bsdnet by change `/etc/sysctl` you mean apply the above sysctls or are you referring to some other sysctls?

@karan why do we need to enable sysrq? I don't think it's required for kdump, unless we want to trigger kdump as a test by forcing a panic via sysrq.

Reply:
Oh I misread the sysctl comment as sysrq. Disregard.